frontend/src/static/js/v_summary_charts.js

```diff
@@ -1,94 +1,3 @@
-window.dismissTab = function dismissTab(node) {
-  const parent = node.parentNode;
-  parent.style.display = 'none';
-};
-
-window.comparator = (fn, sortingOption = '') => function compare(a, b) {
-  let a1;
-  let b1;
-  if (sortingOption) {
-    a1 = fn(a, sortingOption).toLowerCase
-        ? fn(a, sortingOption).toLowerCase()
-        : fn(a, sortingOption);
-    b1 = fn(b, sortingOption).toLowerCase
-        ? fn(b, sortingOption).toLowerCase()
-        : fn(b, sortingOption);
-  } else {
-    a1 = fn(a).toLowerCase ? fn(a).toLowerCase() : fn(a);
-    b1 = fn(b).toLowerCase ? fn(b).toLowerCase() : fn(b);
-  }
-  if (a1 === b1) {
-    return 0;
-  } if (a1 < b1) {
-    return -1;
-  }
-  return 1;
-};
-
-// ui funcs, only allow one ramp to be highlighted //
-let drags = [];
-
-function getBaseTarget(target) {
-  return (target.className === 'summary-chart__ramp')
-      ? target
-      : getBaseTarget(target.parentElement);
-}
-
-function deactivateAllOverlays() {
-  document.querySelectorAll('.summary-chart__ramp .overlay')
-      .forEach((x) => {
-        x.className = 'overlay';
-      });
-}
-
-function dragViewDown(evt) {
```

Comment:
for all the changes in this file except `window.comparator`: Would it be better to move it to the component definition? Since this isn't really a utility-like function and relies heavily on `summary`'s structure. 
Likewise for `dismissTab`, but in that case it would be moving it to `summary` instead of `summary_chart`'s definition

Reply:
all moved into respective component except deactivatealloverlay which is used in both summary and summarycharts

Reply:
all moved into respective component except deactivateAllOverlay which is used in both summary and summaryCharts